### ISSUE TITLE
@Transactional should not override @RunOnVirtualThread in reasteasy reactive

### ISSUE THREAD

**Issue by imperatorx** [quarkusio/quarkus#34319]
### Describe the bug

PR #24942, as choice of caution, disables virtual threads when `@Transactional` is added to a reasteasy reactive method. Narayana will soon (7.0) have support for virtual threads (https://github.com/jbosstm/narayana/pull/2077), so I think this restriction should go away for those who want to experiment with or build on this. 

With narayana built from trunk and a loom friendly jdbc driver, @RunOnVirtualThread works if the method is not annotated with `@Transactional`, but there is programmatic transaction control inside using `QuarkusTransaction.requiringNew()`.

### Expected behavior

```java
@Transactional
@RunOnVirtualThread
public String foo() {
  // access SQL datasource or cusom XA resouce here works, transaction is present
  // is executed on a virtual thread
}
```

### Actual behavior

```java
@Transactional
@RunOnVirtualThread
public String foo() {
  // access SQL datasource or cusom XA resouce here works, transaction is present
  // is executed on the main thread pool instead of a virtual thread
}
```

### Output of `java -version`

19

### Quarkus version or git rev

3.1.2.Final

**Comment by geoand** [comment]
cc @cescoffier @franz1981

**Comment by franz1981** [comment]
AFAIK narayana is now pinnnig the virtual thread; maybe a fix is in flight for that.

**Comment by geoand** [comment]
That is what I think the issue is saying

**Comment by imperatorx** [comment]
> AFAIK narayana is now pinnnig the virtual thread; maybe a fix is in flight for that.

It is, I made the pr:
> Narayana will soon (7.0) have support for virtual threads (https://github.com/jbosstm/narayana/pull/2077),

**Comment by cescoffier** [comment]
Yes, it got fixed last week. So we can finally improve.

Do you know when we would be able to update narayana? Because that's what will drive the change.

**Comment by imperatorx** [comment]
> Do you know when we would be able to update narayana? Because that's what will drive the change.

They said it will be in 7.x as public methods got their synchronized flags removed (public API change)

**Comment by imperatorx** [comment]
Also, what are the plans regarding virtual threads and classic resteasy integration? It seems virtual threads are a better fit for blocking (non-reactive) semantics.

**Comment by mmusgrov** [comment]
The plan is to release Narayana with @imperatorx 's enhancement this week.

**Comment by imperatorx** [comment]
`7.0.0.Final` was released: https://search.maven.org/search?q=g:org.jboss.narayana.jta

**Comment by mmusgrov** [comment]
The quarkus [component upgrade PR](https://github.com/quarkusio/quarkus/pull/34781) has nearly completed its checks.

**Comment by cescoffier** [comment]
I just merged the narayana update - will lift the limitation tomorrow

**Comment by cescoffier** [comment]
Fixed with #34808
